### ISSUE TITLE
Enemy unit exits captured WF 

### ISSUE THREAD

**Issue by ugotbroiled** [OpenRA/OpenRA#14219]
In RA, when you capture an enemy WF that is currently has a unit ready, it comes out of the newly captured WF as an enemy unit, which then can kill the engineer that just captured it.  

The production must already be in progress when the capture begins and it must finish before the capture ends.  The unit will wait until the capture is complete and then exit the WF.

![screen shot 2017-10-19 at 3 01 08 pm](https://user-images.githubusercontent.com/12901045/31788943-6606905e-b4de-11e7-9880-738d74c0bf08.png)

This could also affect other production facilities such as barracks, airfields, etc when there is only 1 of those type of production facilities and the unit has no alternate building to exit.

**Comment by FrameLimiter** [comment]
Duplicate of #13983

**Comment by AndriiYukhymchak** [comment]
fixing it

**Comment by netnazgul** [comment]
Some thoughts [here](https://github.com/OpenRA/OpenRA/pull/14220#issuecomment-338113313)
